lib/queue.js

```diff
@@ -749,9 +698,17 @@ class Queue extends Emitter {
 
   _checkStalledJobs(interval, cb) {
     const promise = this._doStalledJobCheck();
-    if (cb) helpers.asCallback(promise, cb);
+    // TODO: what DO
+    // TODO: reintroduce tests
+    if (cb) {
+      promise.then(
+        (value) => process.nextTick(() => cb(null, value)),
+        (err) => process.nextTick(() => cb(err))
+      );
+    }
```

Comment:
So this is sad, and I would like to find another solution. The `checkStalledJobs` method supports a callback so that the library user may receive callbacks for _all_ attempted stall checks - not just the first. Removing the callback doesn't permit us to propagate any useful information, as the process won't really finish until `Queue#close` is called.

Perhaps we use events and emit an event? How do we handle errors in that case?